src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/model/CaseRecord.java

```diff
@@ -0,0 +1,19 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model;
+
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model.Envelope;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+public interface CaseRecord extends CaseDataCreator {
+
+    @FunctionalInterface
+    interface Construct {
+        CaseDataCreator apply(Envelope envelope, CaseDetails caseDetails);
```

Comment:
Why not just a bi-Function ?
Its recommended to use the standard java  functional interfaces